### PR TITLE
[SC-597] Feat: Add Inverter Reverter to Upgrade Mechanism

### PR DESCRIPTION
**Pull request by marvinkruse**
Work from [585](https://github.com/InverterNetwork/contracts/pull/585) but applied to `dev` instead of `hats-fixes`.

### CODE REVIEW

**Comment by marvinkruse** [comment]
@0xNuggan @Zitzak I already reviewed the original PR and was fine with it - but if someone of you can just generally quickly skim over it to see if it's all good that would be good. It's not a Hats fix though, so can happen afterwards/next week if there is no time.

File: script/external/DeployTransactionForwarder_v1.s.sol [hunk 1]
```diff
@@ -20,8 +18,8 @@ contract DeployTransactionForwarder_v1 is Script {
     uint deployerPrivateKey = vm.envUint("ORCHESTRATOR_ADMIN_PRIVATE_KEY");
     address deployer = vm.addr(deployerPrivateKey);
 
-    /// @notice Creates the implementation, beacon and proxy of the transactionForwarder
-    /// @return implementation The implementation of the TransactionForwarder_v1 that will be referenced in the beacon
+    /// @notice Creates the implementatio of the transactionForwarder
```
Comment (Zitzak) [inline script/external/DeployTransactionForwarder_v1.s.sol]:
```suggestion
    /// @notice Creates the implementation of the transactionForwarder
```

**Review by Zitzak — APPROVED** [review]
LGTM. Looked through all the implementation and script, skimmed the tests. Found one typo only but will approve already

**Comment by marvinkruse** [comment]
> LGTM. Looked through all the implementation and script, skimmed the tests. Found one typo only but will approve already

Thank you sir 🙏

**Pull request merged**